app/main.py

```diff
@@ -19,15 +36,22 @@ def extract_content(request: Request, url: str) -> HTMLResponse:
     response = requests.get(url, headers=headers)
     soup = BeautifulSoup(response.content, 'html.parser')
     images = []
-    for img in soup.find_all('img'):
+    headlines = []
+    for img in soup.find_all('img')[:n_images]:
```

Comment:
Put all images in one array so that this works. Then use multi threading `workers=2`